src/inmanta/deploy/work.py

```diff
@@ -243,31 +259,41 @@ def queue_put_nowait(
         task: tasks.Task,
         *,
         priority: TaskPriority,
-        priority_tiebreaker: Optional[int] = None,
+        requested_at: Optional[int] = None,
         reason: Optional[str] = None,
     ) -> None:
         """
         Add a new task to the associated agent's queue.
 
         :param task: The task to queue.
         :param priority: The priority to queue the task at.
-        :param priority_tiebreaker: The tiebreaker value for comparing this task's priority equal-priority tasks.
-            Should only be set if this task was previously dropped from the agent queues, in order to insert it back
-            at its previous position.
+        :param requested_at: Integer value representing when this task was initially requested for the given priority (e.g. if
+            it was deferred for some time. Serves as tiebreaker for comparing this task with equal-priority tasks. If none, the
+            next value in line is used, resulting in the task being added at the end of its priority queue. A value can be
+            reserved (for adding to the queue in the future) through the reserve_requested_at() method.
         """
-        priority_tiebreaker = priority_tiebreaker if priority_tiebreaker is not None else self._entry_count
+        requested_at = requested_at if requested_at is not None else self._entry_count
```

Comment:
```suggestion
        requested_at = requested_at if requested_at is not None else self.reserve_requested_at()
```

Reply:
Actually, this line should go in its entirety. I moved this further down and forgot to remove this.